pkg/controller/elasticsearch/driver/pvc_expansion_test.go

```diff
@@ -0,0 +1,632 @@
+				actualSset:   withClaims(sampleSset, sampleClaim, sampleClaim2),
+			},
+			want:    false,
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := needsRecreate(tt.args.expectedSset, tt.args.actualSset)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("needsRecreate() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if got != tt.want {
+				t.Errorf("needsRecreate() got = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_isStorageExpansion(t *testing.T) {
+	type args struct {
+		expectedSize *resource.Quantity
+		actualSize   *resource.Quantity
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    bool
+		wantErr bool
+	}{
+		{
+			name: "expected == actual: false",
+			args: args{
+				expectedSize: resource.NewQuantity(1, resource.DecimalSI),
+				actualSize:   resource.NewQuantity(1, resource.DecimalSI),
+			},
+			want: false,
+		},
+		{
+			name: "expected > actual: true",
+			args: args{
+				expectedSize: resource.NewQuantity(2, resource.DecimalSI),
+				actualSize:   resource.NewQuantity(1, resource.DecimalSI),
+			},
+			want: true,
+		},
+		{
+			name: "expected < actual: error out",
+			args: args{
+				expectedSize: resource.NewQuantity(1, resource.DecimalSI),
+				actualSize:   resource.NewQuantity(2, resource.DecimalSI),
+			},
+			want:    false,
+			wantErr: true,
+		},
+		{
+			name: "expected is nil",
+			args: args{
+				expectedSize: nil,
+				actualSize:   resource.NewQuantity(1, resource.DecimalSI),
+			},
+			want: false,
+		},
+		{
+			name: "actual is nil",
+			args: args{
+				expectedSize: resource.NewQuantity(1, resource.DecimalSI),
+				actualSize:   nil,
+			},
+			want: false,
+		},
+		{
+			name: "expected and actual are nil",
+			args: args{
+				expectedSize: nil,
+				actualSize:   nil,
+			},
+			want: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := isStorageExpansion(tt.args.expectedSize, tt.args.actualSize)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("isStorageExpansion() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if got != tt.want {
+				t.Errorf("isStorageExpansion() got = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_recreateStatefulSets(t *testing.T) {
+	controllerscheme.SetupScheme()
+	es := func() *esv1.Elasticsearch {
+		return &esv1.Elasticsearch{ObjectMeta: metav1.ObjectMeta{Namespace: "ns", Name: "es", UID: "es-uid"}, TypeMeta: metav1.TypeMeta{Kind: "Elasticsearch"}}
+	}
+	withAnnotation := func(es *esv1.Elasticsearch, key, value string) *esv1.Elasticsearch {
+		if es.Annotations == nil {
+			es.Annotations = map[string]string{}
+		}
+		es.Annotations[key] = value
+		return es
+	}
+
+	sset1 := &appsv1.StatefulSet{ObjectMeta: metav1.ObjectMeta{Namespace: "ns", Name: "sset1", UID: "sset1-uid"}}
+	sset1Bytes, _ := json.Marshal(sset1)
+	sset1Json := string(sset1Bytes)
```

Comment:
cosmetic nit: I [think](https://github.com/golang/go/wiki/CodeReviewComments#initialisms) it is either `JSON` or `json`
```suggestion
	sset1JSON := string(sset1Bytes)
```